pyiceberg/expressions/visitors.py

```diff
@@ -906,7 +906,16 @@ def visit_bound_predicate(self, predicate: BoundPredicate[L]) -> BooleanExpressi
 
 
 def translate_column_names(expr: BooleanExpression, file_schema: Schema, case_sensitive: bool) -> BooleanExpression:
-    return visit(expr, _ColumnNameTranslator(file_schema, case_sensitive))
+    try:
+        return visit(expr, _ColumnNameTranslator(file_schema, case_sensitive))
+    except ValueError as e:
+        if "Not found in file schema" in str(e):
+            if isinstance(expr, BoundIsNull):
+                return AlwaysTrue()
+            else:
+                return AlwaysFalse()
```

Comment:
Curious, why not do this in the visitor itself (_ColumnNameTranslator#visit_bound_predicate)? Is it something we would want decoupled from it?

Reply:
I actually like that more, great suggestion 👍